.github/workflows/touchdown-pipeline/touchdown-build.yml

```diff
@@ -23,73 +23,82 @@ extends:
+                    # - Script to push changes
+                    # ------------------------------------------------------------------------------------------------
 
-              - task: TouchdownBuildTask@1
-                inputs:
-                  environment: 'PRODEXT'
-                  teamId: '35669'
-                  authType: 'OAuth'
-                  authId: 'fa441387-cab4-4f61-969d-fd23388fc7b4'
-                  authKey: $(AUTH_KEY)
-                  isPreview: false
-                  resourceFilePath: 'packages/react-composites/src/composites/localization/locales/en-US/strings.json'
-                  outputDirectoryRoot: 'packages/react-composites/src/composites/localization/locales'
-                  cultureMappingType: 'None'
-                  gitAction: 'COMMIT'
-                displayName: 'Call Touchdown build task for composites and commit changes'
-                condition: eq(variables['commit'], 'true')
+                    $teamProject = $env:SYSTEM_TEAMPROJECT
+                    $repoName= $env:Build_Repository_Name
+                    $pipelineName = $env:Build_DefinitionName
+
+                    Write-host "Azure DevOps organization: $env:System_TeamFoundationCollectionUri"
+                    Write-host "Team Project: $teamProject"
+                    Write-host "Repo Name: $repoName"
+                    Write-host "Source Branch: $(Build.SourceBranch)"
+
+
+                    if ("$(git status)".Contains("nothing to commit")) {
+                        Write-Host "Nothing to commit, exiting immediately"
+                        Exit
+                    }
+
+                    #****************************************
+                    #************Configure GIT***************
+                    #****************************************
+                    Write-Host "##[group] Configure GIT"
+
+                    #Setting Global Git Username and Email. NOTE: Branch will be created as this user
+                    git config --global user.email "79475487+mgamis-msft@users.noreply.github.com"
+                    git config --global user.name "mgamis-msft"
+
+                    #**********************************************
+                    #***Checkout current branch and push changes***
+                    #**********************************************
+
+                    Write-Host "##[group] Checkout current branch and push changes"
+                    #Create Branch and add changes
+
+                    Write-Host "Checkout current branch"
+
+                    $branchName = "$(Build.SourceBranch)"
+                    $branchName = $branchName -replace "refs/heads/", ""
+
+                    git remote set-url origin https://github.com/$repoName
+                    git checkout $branchName
+                    git add -A
+                    git commit -m "TDBuild latest localization" -a
+                    git push origin $branchName

```

Comment:
Out of interest, how does this task have access to the repo? (is it because it is triggered from the repo it gets given a token by github?)

Reply:
(that's how our regular github actions work)

Reply:
I'm not quite sure. But I think you are right, because it is triggered from the repo.